tpchgen-cli/src/runner.rs

```diff
@@ -0,0 +1,208 @@
+//! [`PlanRunner`] for running [`OutputPartitionPlan`]s.
+
+use crate::csv::*;
+use crate::generate::{generate_in_chunks, Source};
+use crate::parquet::generate_parquet;
+use crate::plan::{OutputLocation, OutputPartitionPlan};
+use crate::tbl::*;
+use crate::tbl::{LineItemTblSource, NationTblSource, RegionTblSource};
+use crate::{OutputFormat, Table, WriterSink};
+use log::{debug, info};
+use std::io;
+use std::io::BufWriter;
+use tpchgen::generators::{
+    CustomerGenerator, LineItemGenerator, NationGenerator, OrderGenerator, PartGenerator,
+    PartSuppGenerator, RegionGenerator, SupplierGenerator,
+};
+use tpchgen_arrow::{
+    CustomerArrow, LineItemArrow, NationArrow, OrderArrow, PartArrow, PartSuppArrow,
+    RecordBatchIterator, RegionArrow, SupplierArrow,
+};
+
+#[derive(Debug)]
+pub struct PlanRunner {
```

Comment:
I moved the code that actually creates the plan to this module -- and I have a plan to orchestrate the creation here.